crates/sargon/src/keys_collector/host_interaction/key_derivation_request.rs

```diff
@@ -0,0 +1,26 @@
+use crate::prelude::*;
+
+/// A collection of derivation paths, on a per-factor-source basis.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct KeyDerivationRequest {
+    pub per_factor_source:
```

Comment:
maybe a note that this `IndexMap` will in most cases contain a single Key-value, and only for DeviceFactorSource kind do we ever put more than one key value (poly)